packages/neon_framework/packages/account_repository/lib/src/utils/http_client_builder.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:account_repository/src/models/models.dart';
+import 'package:http/http.dart' as http;
+import 'package:neon_framework/storage.dart';
+import 'package:neon_http_client/neon_http_client.dart';
+import 'package:nextcloud/nextcloud.dart';
+
+/// Builds a [NextcloudClient] authenticated with the given [credentials].
+NextcloudClient buildClient({
+  required http.Client httpClient,
+  required String userAgent,
+  required Credentials credentials,
+}) {
+  final cookieStore = NeonStorage().cookieStore(
+    accountID: credentials.id,
+    serverURL: credentials.serverURL,
+  );
+
+  final neonClient = NeonHttpClient(
+    cookieStore: cookieStore,
+    userAgent: userAgent,
+    client: httpClient,
+    timeLimit: kDefaultTimeout,
+    baseURL: credentials.serverURL,
+  );
+
+  return NextcloudClient(
+    credentials.serverURL,
+    loginName: credentials.username,
+    password: credentials.password,
+    appPassword: credentials.password,
+    httpClient: neonClient,
+  );
+}
+
+/// Builds an unauthenticated [NextcloudClient] for the given [serverURL].
+NextcloudClient buildUnAuthenticatedClient({
+  required http.Client httpClient,
+  required String userAgent,
+  required Uri serverURL,
+}) {
+  final neonClient = NeonHttpClient(
```

Comment:
```suggestion
  final httpClient = NeonHttpClient(
```

Reply:
This is already the parameter name of the http client backing the neon client.